datastream-kafka/src/main/java/com/linkedin/datastream/kafka/KafkaProducerWrapper.java

```diff
@@ -166,6 +166,11 @@ private void populateDefaultProducerConfigs() {
 
   private Optional<Producer<K, V>> maybeGetKafkaProducer(DatastreamTask task) {
     Producer<K, V> producer = _kafkaProducer;
+    if (!_tasks.contains(task)) {
```

Comment:
Does adding a comment here make sense, that if the task is not assigned, production should be disallowed?

Reply:
I feel the code is self explainatory.